cms/templates/js/unit-access-editor.underscore

```diff
@@ -0,0 +1,133 @@
+<%
+var groupAccess = xblockInfo.get('group_access');
+var selectablePartitions = [];
+var groupIdToName = {};
+
+contentGroups.forEach(function (partition) {
+    if (partition['scheme'] != 'cohort' && partition['scheme'] != 'enrollment_track') {
+        return;
+    }
+    if (partition['scheme'] == 'enrollment_track') {
+        var tracks = [];
+        var trackSelected = false;
+        partition['groups'].forEach(function (group) {
+            if (group['selected']) {
+                trackSelected = true;
+            }
+        });
+        // Only add enrollment tracks if there are multiple tracks to restrict from
+        if (partition['groups'].length < 2 && !trackSelected) {
+            return;
+        }
+    }
+    selectablePartitions.push(partition);
+});
+
+selectablePartitions.forEach(function (partition) {
+    var schemeId = partition['id'];
+    var groups = partition['groups'];
+    groups.forEach(function (group) {
+        var groupId = group['id'];
+        var groupName = group['name'];
+        var fullId = schemeId.toString() + '-' + groupId.toString();
+        groupIdToName[fullId] = groupName;
+    })
+});
+
+var deletedGroups = [];
+
+for (var schemeId in groupAccess) {
+    groupAccess[schemeId].forEach(function (groupId) {
+        var fullId = schemeId.toString() + '-' + groupId.toString();
+        if (!groupIdToName[fullId]) {
+            deletedGroups.push(fullId);
+        }
+    })
+}
+
+// Need to include deleted groups logic to not change is_active boolean in contentstore/utils.py get_user_partition_info
+deletedGroups.forEach(function (fullId) {
+    var schemeId = fullId.split('-')[0];
+    var groupId = fullId.split('-')[1];
+    var partitionExists = false;
+    selectablePartitions.forEach(function (partition) {
+        if (partition['id'] === schemeId) {
+            partitionExists = true;
+            partition['groups'].push(
+                {
+                    "deleted": true,
+                    "selected": true,
+                    "id": groupId,
+                    "name": ""
+                }
+            );
+        }
+    })
+
+    if (!partitionExists) {
+        selectablePartitions.push(
+            {
+                "scheme": "cohort",
+                "id": schemeId,
+                "groups": [
+                    {
+                        "deleted": true,
+                        "selected": true,
+                        "id": groupId,
+                        "name": "Deleted Group"
+                    }
+                ],
+                "name": "Content Group Configuration"
+            }
+        )
+    }
+})
+%>
+<form>
+<h3 class="modal-section-title">
+    <%- gettext('Unit Access') %>
+</h3>
+<div class="modal-section-content access-change">
+    <label for="partition-select" class="group-select-title"><%- gettext('Restrict access to:') %>
```

Comment:
Nit: don't need "for" here because the nesting does the job. This means you can also get rid of the ID altogether and just use the user-partition-select class for selectors.